articles/how-to-use-an-api-in-a-vuejs-project/index.md

```diff
@@ -0,0 +1,367 @@
+
+**Axios Installation**
+
+To use Axios on your project, you need to first install it. This can be done in two ways:
+
+-  By using `npm`; a standard package manager for the JavaScript runtime environment Node.js. You can now see why having Node.js installed on your computer was a prerequisite.
+- By using `yarn`; a package manager that also acts as a project manager. It is synergetic with the npm registry and has the same features. To install yarn in your project, paste the following line of code in your terminal `npm install --global yarn`
+
+
+With npm:
+
+```
+npm i axios
+```
+With yarn:
+
+```
+yarn add axios
+```
+Next, you import Axios in your `src/main.js` file
+
+```
+import axios from 'axios';
+Vue.prototype.$http = axios;
+```
+
+**How to make an API request and display data using Axios.**
+
+We will make the API request using a `get` method. A get method is used to call data from a resource, the resource in this context is our API.  We want this API request running synchronously therefore, we use a promise-based function with keywords [async/await](https://github.com/section-engineering-education/engineering-education/pull/2235). You may wonder why we used a promise-based function, this is because a promise provides a way of monitoring the state of the API and acts as a container for the data that will eventually return.
+However, we also need to test for errors using the [try/catch](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch) method. `try` is used to check for errors while `catch` is used to handle all errors that are detected.
+
+
+Copy the code below to your `App.vue` file
+
+```
+<template></template>
+
+<script>
+export default {
+data()  {
+posts: [ ]
+
+},
+
+methods: {
+async getData () {
+     try {
+              const response = await this.$http.get('http://jsonplaceholder.typicode.com/posts');
+              // JSON responses are automatically parsed.
+              this.posts = response.data
+           }
+          catch (error) {
+             console.log(error);
+         }
+
+       }
+   }
+}
+</script>
+```
+
+The above block of code in the `methods` property will be explained line by line. 
+
+```
+async getData (){
+```
+
+A function named `getData()` is created . It is in this function the API will be called.
+The `async` keyword is prepended on the `getData` function to show that the function will return a promise.
+
+```
+try {
+    const response = await this.$http.get('http://jsonplaceholder.typicode.com/posts');
+```
+
+`try` property defines a block of code to be tested for errors as the code is executed. In the block of code `const response = await this.$http.get('http://jsonplaceholder.typicode.com/posts');`, a get request is made with the `get` keyword using axios i.e.`$http` to get data from the URL. `await` is prepended to the request because we are calling a promise. The request is stored in a variable called `response`.
```

Comment:
We are not calling a promise. The get function will return a promise. The request is not stored in the variable response. The data returned from the API after the promise is resolved will be stored in the variable response.